ddf.py

```diff
@@ -63,26 +69,33 @@ def __hash_string(self, str):
         return hasher.hexdigest()
 
     def __get_sub_directories(self, directory):
-        return [os.path.join(directory, o)
-            for o in os.listdir(directory)
-                if os.path.isdir(os.path.join(directory,o))]
-
+        sub_dir_list = [] 
+        for o in os.listdir(directory): 
+            if os.path.isdir(directory):
+                sub_dir_list.append(os.path.join(directory,o))
+        return sub_dir_list
+#   Get feedback on why this doesn't work...
+#    def __get_files(self, directory):
+#        files_list = []
+#        for o in os.listdir(directory):
+#            if os.path.isfile(o):
+#                files_list.append(os.path.join(directory,o))
+#        return files_list
+#
     def __get_files(self, directory):
-        return [os.path.join(directory, o)
-            for o in os.listdir(directory)
-                if os.path.isfile(os.path.join(directory,o))]
+        return [os.path.join(directory, o) for o in os.listdir(directory) if os.path.isfile(os.path.join(directory,o))]
 
-    def __retry_oserror(self, func, *args):
+    def __retry_oserror(self, passed_func, *args):
```

Comment:
I prefer the more terse name `func` since it is plain that it's a function reference that's been passed as an argument to this method.